Minor optimizations

juliyvchirkov opened the pull request:
Minor optimizations to move the whole routine of hex colors declaration to the initial block

The idea is to have one and only declaration per color, stored in variable and reused elsewhere in the script late

This way It would be cleaner and easier to debug the template, the proposed refactoring dropes out the duplicates of color hex values for iTerm2 and thus delivers template optimization and unification in general

juliyvchirkov committed:
Minor optimizations

Minor optimizations to move the whole routine of hex colors declaration to the initial block

The idea is to have one and only declaration per color, stored in variable and reused elsewhere in the script late

This way It would be cleaner and easier to debug the template, the proposed refactoring dropes out the duplicates of color hex values for iTerm2 and thus delivers template optimization and unification in general

juliyvchirkov committed:
Making the patch less shell-specific

Previous patch edition was too bash-specific - in-variable search-n-replacement `${var////}` pattern works in bash since v4, but I'm not sure if other shells will overcome it

This update will achieve the goal despite of a shell

nvllsvm commented on templates/default.mustache:
```diff
@@ -78,13 +78,13 @@ put_template 21 $color21
-  put_template_custom Pk {{base05-hex}} # selected text color
-  put_template_custom Pl {{base05-hex}} # cursor
-  put_template_custom Pm {{base00-hex}} # cursor text
+  put_template_custom Pg $(sed -e 's#/##g' <<<$color07) # foreground
```
This script uses the `/bin/sh` shebang, so it shouldn't use a bashism like `<<<`. It should consist solely of POSIX-compliant shell to keep compatibility with the most shells. Ex. dash is a POSIX-only shell.

nvllsvm commented on templates/default.mustache:
```diff
@@ -78,13 +78,13 @@ put_template 21 $color21
-  put_template_custom Pk {{base05-hex}} # selected text color
-  put_template_custom Pl {{base05-hex}} # cursor
-  put_template_custom Pm {{base00-hex}} # cursor text
+  put_template_custom Pg $(sed -e 's#/##g' <<<$color07) # foreground
```
Using `sed` is less efficient as it's an external process. Your changes result in 7 external processes being spawned each time a color file is sourced.

juliyvchirkov replied:
Thanks for notice on bash / sh, I've indeed missed that point. Also  I supposed ` sed ` is ok since it is already utilized on line 43, but anyway I've refused ` sed ` and bashism, and restyled the update in pure sh

juliyvchirkov committed:
sh way

Thanks for notice on bash / sh, I've indeed missed that point. Also  I supposed ` sed ` is ok since it is already utilized on line #43, but anyway I've refused ` sed ` and bashism, and restyled the update in pure sh

juliyvchirkov commented:
Thanks for notice on bash / sh, I've indeed missed that point. Also  I supposed ` sed ` is ok since it is already utilized on line chriskempson#43, but anyway I've refused ` sed ` and bashism, and restyled the update in pure sh
